### PR TITLE
🏗 Standardize interface for CI modules

### PR DESCRIPTION
**Pull request by rcebulko**
I came across these files in a PR review (turns out the diff was a mistake based on rebasing, so they weren't part of the PR) but noticed a chance to unify some of the CI types here. I didn't realize it was a mistake until after I'd commented and thought about it a bit, so I figured it was worth putting in a PR for you to peek at

This standardizes the common function names across CI types (`isPushBuild`, `isPullRequestBuild`, `pullRequestBranch`, and `pullRequestSha` with the goals of a) calling code more general to allow easier transition between CI and b) simplify shared logic within ci.js. I haven't updated any calling code, but PTAL and let me know what you think of this structure.

### CODE REVIEW

File: build-system/common/ci.js [hunk 1]
```diff
@@ -87,49 +48,36 @@ function isCiBuild() {
...
-  ciPullRequestSha,
   isCiBuild,
   isPullRequestBuild,
   isPushBuild,
+
+  isTravisBuild: testBuildType('travis'),
+  isGithubActionsBuild: testBuildType('github_actions'),
+  isCircleciBuild: testBuildType('circleci'),
+
+  ciPullRequestBranch: getPrBuildVar(pullRequestBranch),
```
Comment (rcebulko) [inline build-system/common/ci.js:81]: These could be expanded pretty easily to support fields like build/job URL and commit sha across the different build types

**Review by rsimha — COMMENTED** [review]
Thanks for starting this discussion.

My initial thoughts:
- Simplifying function lookup would be nice, but I don't think dynamic requires of the same functions from different files is the way to go
- Re: unifying functions across services, it turns out there are non-trivial differences. E.g. Travis separates `TRAVIS_BRANCH` from `TRAVIS_PULL_REQUEST_BRANCH`, while GH actions has `GITHUB_REF` (the prefix needs to be removed), and CircleCI just has `CIRCLE_BRANCH`. There are more such instances.
- I've started documenting these differences in a new PR with different sets of functions per service, unified in `ci.js`
- In future, some tasks may have to use a combination of service-specific and service-agnostic functionality, which will be tricky if all files export the same function names

TL;DR: I don't think this PR can be used exactly as written, but some optimization ideas could be applicable. I'll loop you in on those PRs.

SGTY? 😃

**Comment by rsimha** [comment]
As promised, a new PR has been sent to you for review: #31527. 

Highlights relevant to this PR:
- The lookup table and the long list of require statements have been fully removed
- Service-level differences are now self-documented by virtue of being co-located in `ci.js`
- The only service-specific functions still necessary are `isTravisBuild()`, `isGithubActionsBuild()`, and `isCircleciBuild()`, which now live in `ci.js`

TL;DR: The initial explorations in this PR were useful. Thank you!
